unison-core/src/Unison/Term.hs

```diff
@@ -171,8 +172,8 @@ bindSomeNames
 --                   || traceShow e False
 --                   = undefined
 bindSomeNames ns e = bindNames keepFree ns e where
-  keepFree = Set.difference (freeVars e)
-                            (Set.map Name.toVar $ Rel.dom (Names.terms0 ns))
```

Comment:
I thought I had an answer for this before, but I wonder if there could be any synthetic names like `Symbol 3 foo` that _are_ supposed to be kept free, but would now wouldn't be, since we don't consider the `id` in the new `Set.difference`?

Good question for @pchiusano.

Reply:
@aryairani Good question. I think you've identified how the behavior might have changed here (but note the `id` _is_ considered in the new `Set.difference` - it's in `Symbol`'s `Ord` instance.

Before:

```haskell
{ Symbol 1 "foo" } `difference` { Name.toVar "foo1" }
{ Symbol 1 "foo" } `difference` { Symbol 0 "foo1 }
{ Symbol 1 "foo" }
```

After:

```haskell
{ Name.fromVar (Symbol 1 "foo") } `difference` { "foo1" }
{ "foo1" } `difference` { "foo1" }
{ }
```

😢